src/wayland.rs

```diff
@@ -246,8 +250,22 @@ impl Wayland {
                     zwlr_layer_shell_v1::Layer::Top,
                     "waycorner".to_owned(),
                 );
-                let size = corner_config.size.into();
-                layer_surface.set_size(size, size);
+                
+                let size_width = 
+                	if corner_config.size_width != 10 {
```

Comment:
If `size_width` is explicitly set to `10` and `size` is set to `50`, it will use `50` instead of 50. This could be solved by using the custom type I suggested above or by using `Option` for `size_width/height`. My preference goes to the custom type.